tic_tac_toe/game.py

```diff
@@ -87,18 +137,63 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
-
 
+    if get_winner(game) != None:
+        raise InvalidMovement("Game is over")  
+
+    if _board_is_full(game['board']) and get_winner(game) == None:
+        raise InvalidMovement("Game is over.")  
+
+    if game['next_turn'] == player:
+        pass
+    else:
+        raise InvalidMovement("\"{}\" moves next".format(game['next_turn']))
+  
+ 
+    if not _position_is_valid(position):
+        raise InvalidMovement("Position out of range.")
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement("Position already taken.")
+#    if get_next_turn(game) != player:
+#    raise InvalidMovement("{} moves next".format(player))
+    
+    x, y = position            
+    game['board'][x][y] = player
+   
+    if _board_is_full(game['board']) and get_winner(game) == None:
+            raise GameOver("Game is tied!")   
+    
+    if get_winner(game) != None:
+        raise GameOver("\"{}\" wins!".format(game['winner']))
+        
+    if player == "X":
+        game['next_turn']="O"
+    else:
+        game['next_turn']="X"
+    
+        
+            
+
+    
+    
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    rep = ''
+    i = 0 
+    for row in game['board']:
+        rep+=('  |  '.join(row)) + '\n'
+        if i<2:
+            rep+=('--------------\n') 
+            i+=1
+    return '\n' + rep
 
 
 def get_next_turn(game):
     """
     Returns the player who plays next, or None if the game is already over.
     """
-    pass
+    if not _board_is_full(game['board']):
+        return game['next_turn']
+    return None
```

Comment:
Nice code overall. You don't need to call return None explicitly because that will be the default return value of a function when there is no return statement. Try to avoid writing long lines, you can put long statements into multiple lines by using "\" at the end of the line.
That's about all I can find.